crates/bevy_render/src/texture/hdr_texture_loader.rs

```diff
@@ -23,6 +24,10 @@ pub enum HdrTextureLoaderError {
     Io(#[from] std::io::Error),
     #[error("Could not extract image: {0}")]
     Image(#[from] image::ImageError),
+    #[error("Could not convert total number of bytes in decoded image from u64 to usize: {0}")]
+    TryIntoTotalBytes(#[from] std::num::TryFromIntError),
+    #[error("Could not return a 32bit RGB image reference")]
+    ImageReferenceRgb32F,
```

Comment:
I would wait for other opinions before acting on this, but IMO this could be removed and replaced with a
```rust
.expect("Image format should be Rgb32F")
```
There should be no way for [`HdrDecoder`](https://github.com/image-rs/image/blob/4594c80b3626076af128b7a853c8e5c5187f7a64/src/codecs/hdr/decoder.rs#L305) to be anything except `Rgba32F` when converted to a `DynamicImage`.